### ISSUE TITLE
Redis URL improvement

### ISSUE THREAD

**Issue by skorokithakis** [timvisee/send#23]
Right now, the Redis address can just be a hostname. It would be better if Send supported the full URL (`redis://username:password@host:port/dbnum`) so we could point it to already-existing Redis servers.

**Comment by timvisee** [comment]
You can also set the port and password through a separate variable: https://github.com/timvisee/send/blob/master/server/config.js#L62-L76

But I agree that supporting the more conventional URL would be better.

**Comment by skorokithakis** [comment]
Oh, that's good enough, thanks! I'm trying to set up Send using Compose at the moment, and this seems adequate.

I don't know if you'd be interested to add support, but I wrote a tool for easier deployment of apps such as this one. It's still early, but it's quite useful to me, at least: https://pypi.org/project/docker-harbormaster/

**Comment by timvisee** [comment]
> Oh, that's good enough, thanks!

Great. I'll leave it open so Redis URL support can be configured.

> I don't know if you'd be interested to add support, but I wrote a tool for easier deployment of apps such as this one. It's still early, but it's quite useful to me, at least: https://pypi.org/project/docker-harbormaster/

Sounds cool. What would 'adding support' mean, having to provide a compose file?

You might find this template useful: https://github.com/timvisee/send-docker-compose

**Comment by skorokithakis** [comment]
Yes, it's just providing a compose file with the `{{ HM_DATA_DIR }}/my_data:/some_data_dir` volumes defined.

Then you could tell people to just add:

```
repositories:
  timvisee-send:
    url: https://github.com/timvisee/send.git
```

to their config and it would be automatically spun up. It's probably too early for me to push it very hard, I was just wondering whether something like this would be interesting to app developers.

**Comment by pirate** [comment]
I added more documentation to the Docker / Docker Compose configs for the `REDIS_HOST`, `REDIS_PORT`, and `REDIS_PASSWORD` options here:

- https://github.com/timvisee/send/pull/34
- https://github.com/timvisee/send-docker-compose/pull/4

**Comment by timvisee** [comment]
I wonder what the more conventional approach would be?

I've mostly seen separate variables, but maybe Redis URLs are used elsewhere. I'd be happy with both.

**Comment by skorokithakis** [comment]
Separate variables are good too, as long as I can specify all of them (host/port/username/pw/db).

**Comment by timvisee** [comment]
I've now added all desired variables: `REDIS_{HOST,PORT,USER,PASSWORD,DB}`.

I've released this as part of `v3.4.11`.

It should be available when this pipeline succeeds: https://gitlab.com/timvisee/send/-/pipelines/305718044
